common/featureset.cpp

```diff
@@ -0,0 +1,56 @@
+#include "featureset.h"
+#include <QMap>
+#include <QDebug>
+
+FeatureSet::FeatureSet()
+{
+
+}
+
+QMap<QString, bool> FeatureSet::getDefaultFeatureList() {
+    initalizeFeatureList(featureList);
+    return featureList;
+}
+
+void FeatureSet::initalizeFeatureList(QMap<QString, bool> &featureList){
+    featureList.insert("client_id", false);
+    featureList.insert("client_ver", false);
+    featureList.insert("feature_set", false);
+}
+
+void FeatureSet::enableRequiredFeature(QMap<QString, bool> &featureList, QString featureName){
+    if (featureList.contains(featureName))
+        featureList.insert(featureName,true);
+}
+
+void FeatureSet::disableRequiredFeature(QMap<QString, bool> &featureList, QString featureName){
+    if (featureList.contains(featureName))
+        featureList.insert(featureName,false);
+}
+
+QMap<QString, bool> FeatureSet::addFeature(QMap<QString, bool> &featureList, QString featureName, bool isFeatureRequired){
+    featureList.insert(featureName,isFeatureRequired);
+    return featureList;
+}
+
+QMap<QString, bool> FeatureSet::identifyMissingFeatures(QMap<QString, bool> featureListToCheck, QMap<QString, bool> featureListToCompareTo){
+    QMap<QString, bool> missingList;
+    QMap<QString, bool>::iterator i;
+    for (i = featureListToCompareTo.begin(); i != featureListToCompareTo.end(); ++i) {
+        if (!featureListToCheck.contains(i.key())) {
+            missingList.insert(i.key(), i.value());
+        }
+    }
+    return missingList;
+}
+
+bool FeatureSet::isRequiredFeaturesMissing(QMap<QString, bool> featureListToCheck, QMap<QString, bool> featureListToCompareTo) {
```

Comment:
Im not certain this is possible (or rather its already doing that).  Or maybe im miss-understanding the usage suggestion of the any_of.  Given the value(s) being compared are a qmap we first need to find out if any of the value(s) are required (which is the bool in the qmap) and then from there determine if any of those true value key's exist in the feature list of the client being passed.  which is what contains does.